synapse/handlers/auth.py

```diff
@@ -914,8 +1047,12 @@ async def validate_login(
         if login_type == LoginType.PASSWORD and self.hs.config.password_localdb_enabled:
             known_login_type = True
 
+            # we've already checked that there is a (valid) password field
+            password = login_submission["password"]
+            assert isinstance(password, str)
```

Comment:
Would it make sense to pull out `password` once at the top of the method?

Reply:
https://github.com/matrix-org/synapse/pull/8849 removes one of the references, so I'm sticking with this for now. (also, you'd still need the assertions to keep mypy happy.)

Reply:
(Instead of needing the comment twice that we already checked and `password` is a valid field.)